janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/optimize/step/JanusGraphPropertiesStep.java

```diff
@@ -189,14 +188,24 @@ public void reset() {
     ===== HOLDER =====
      */
 
-    private final List<HasContainer> hasContainers;
+    private final ArrayList<HasContainer> hasContainers;
     private int limit;
     private final List<HasStepFolder.OrderEntry> orders = new ArrayList<>();
 
 
     @Override
-    public void addAll(Iterable<HasContainer> has) {
-        Iterables.addAll(hasContainers, has);
+    public int getHasContainersAmount() {
+        return hasContainers.size();
+    }
+
+    @Override
+    public void ensureHasContainersCapacity(int size) {
+        hasContainers.ensureCapacity(size);
```

Comment:
Nitpick: This method could also be called `ensureAdditionalHasContainersCapacity(int additionalSize)` and add `hasContainers.size()` on its own. By that, we could avoid the extra method `getHasContainersAmount()`.

Reply:
Changed to `ensureAdditionalHasContainersCapacity` and removed `getHasContainersAmount`